internal/provider/resource_virtual_machine_group_test.go

```diff
@@ -0,0 +1,137 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/jimeh/undent"
+)
+
+func init() { //nolint:gochecknoinits
+	resource.AddTestSweepers("katapult_virtual_machine_group",
+		&resource.Sweeper{
+			Name: "katapult_virtual_machine_group",
+			F:    testSweepVMGroups,
+		})
+}
+
+func testSweepVMGroups(_ string) error {
+	m := sweepMeta()
+	ctx := context.TODO()
+
+	vmgs, _, err := m.Client.VirtualMachineGroups.List(
+		ctx, m.OrganizationRef(),
+	)
+	if err != nil {
+		return err
+	}
+
+	for _, vmg := range vmgs {
+		log.Printf(
```

Comment:
Sweeping everything can be dangerous if the sweeper is ever run against the wrong environment/org. To keep things safe we should only remove resources which have the test-specific prefix (`tf-acc-test-`) defined in the `testAccResourceNamePrefix` variable.

Something like this should do the job:

```suggestion
		if !strings.HasPrefix(vmg.Name, testAccResourceNamePrefix) {
			continue
		}

		log.Printf(
```

This works for tests that use of the `(*testTools) ResourceName(string) string` helper function to produce a name value. The helper adds the test prefix and a randomized suffix to whatever string it is given. This ensures test related resources are easy to spot, and that you can't get name collisions when run in parallel for resources which require unique names.

The Virtual Machines resource sweeper has this prefix check too, even though there's a test which doesn't use the `ResourceName` helper, as it does not specify a name. This is because Virtual Machine resources will receive a generated name if none is specified, by default this generated name is prefixed with `tf-` when using the provider normally. But within the acceptance tests this prefix is set to the test prefix instead, allowing the sweeper to find and destroy it.